validator/eth2-migrate.sh

```diff
@@ -0,0 +1,169 @@
+#!/bin/bash
+
+# TODO:
+# - Implement curl response check
+# - Implement int tests
+# - Implement unit tests
+# - Move slashing_protection data from old location (/root/.eth2validators/validator.db) to new location (/root/.eth2validators/prysm-wallet-v2/validator.db) ??
+
+#############
+# VARIABLES #
+#############
+
+ERROR="[ ERROR-migration ]"
+WARN="[ WARN-migration ]"
+INFO="[ INFO-migration ]"
+
+HTTP_WEB3SIGNER="http://web3signer.web3signer-prater.dappnode:9000"
+NETWORK="prater"
+WALLET_DIR="/root/.eth2validators"
+WALLETPASSWORD_FILE="${WALLET_DIR}/walletpassword.txt"
+BACKUP_DIR="/root/.eth2validators/backup"
+BACKUP_ZIP_FILE="${BACKUP_DIR}/backup.zip"
+BACKUP_KEYSTORES_DIR="${BACKUP_DIR}/keystores" # Directory where the keystores are stored in format: keystore_0.json keystore_1.json ...
+BACKUP_SLASHING_FILE="${BACKUP_DIR}/slashing_protection.json"
+BACKUP_WALLETPASSWORD_FILE="${BACKUP_DIR}/walletpassword.txt"
+
+#############
+# FUNCTIONS #
+#############
+
+# Ensure requirements
+function ensure_requirements() {
+    # Check if web3signer is available: https://consensys.github.io/web3signer/web3signer-eth2.html#tag/Server-Statuss
+    if ! curl -s -X GET \
+    -H "Content-Type: application/json" \
+    --max-time 10 \
+    --retry 5 \
+    --retry-delay 2 \
+    --retry-max-time 40 \
+    "${HTTP_WEB3SIGNER}/upcheck" > /dev/null; then
+        { echo "${ERROR} web3signer not available, manual migration required"; empty_validator_volume; exit 1; }
```

Comment:
Could it be possible that the web3signer is still starting and we have a false negative that deletes all the keys? Maybe it would be good to have a couple of retries.

What happens if the `upcheck` returns false? The curl response will be valid...

Reply:
Yes, there are already a couple of retries

And you right I have to check the curl response and act correspondingly

Reply:
implement on f6161b8d55379c0711699faeb70f14a6839dd066